src/openrct2/ride/Ride.cpp

```diff
@@ -559,22 +556,22 @@ bool track_block_get_next(CoordsXYE* input, CoordsXYE* output, int32_t* z, int32
  *  rct2: 0x006C63D6
  */
 bool track_block_get_previous_from_zero(
-    int16_t x, int16_t y, int16_t z, Ride* ride, uint8_t direction, track_begin_end* outTrackBeginEnd)
+    const CoordsXYZ& startPos, Ride* ride, uint8_t direction, track_begin_end* outTrackBeginEnd)
 {
     uint8_t directionStart = direction;
     direction = direction_reverse(direction);
+    auto trackPos = startPos;
```

Comment:
Why copy this?

Reply:
`trackPos` is updated in the if statement right below. `startPos` is a const-ref.

Reply:
Then why make `startPos` a const-ref in the first place?!

Reply:
That's debatable, I feel like making the interface `const &` by design makes the interface more consistent and we tend to gain more than lose, as most places won't need to alter the object inside. Some cases will require duplication, but then you're most likely just delaying the duplication to the callee instead of the caller.